app/src/main/java/org/p2p/wallet/history/interactor/TransactionsHistoryRepository.kt

```diff
@@ -1,60 +1,66 @@
 package org.p2p.wallet.history.interactor
 
-import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
 import kotlinx.coroutines.withContext
 import org.p2p.solanaj.kits.transaction.SwapDetails
 import org.p2p.solanaj.kits.transaction.TransactionDetails
 import org.p2p.solanaj.model.types.AccountInfo
 import org.p2p.wallet.history.db.dao.TransactionDaoDelegate
 import org.p2p.wallet.history.model.HistoryTransaction
+import org.p2p.wallet.infrastructure.dispatchers.CoroutineDispatchers
 import org.p2p.wallet.infrastructure.network.provider.TokenKeyProvider
 import org.p2p.wallet.rpc.repository.RpcRepository
+import org.p2p.wallet.utils.ifSizeNot
 
-class HistoryTransactionsRepositoryImpl(
+class TransactionsHistoryRepository(
     private val rpcRepository: RpcRepository,
     private val tokenKeyProvider: TokenKeyProvider,
     private val transactionDaoDelegate: TransactionDaoDelegate,
     private val transactionDetailsMapper: TransactionDetailsMapper,
-    private val historyTransactionMapper: HistoryTransactionMapper
-) : HistoryTransactionsRepository {
-    override suspend fun getTransactionsHistory(
+    private val historyTransactionMapper: HistoryTransactionMapper,
+    private val dispatchers: CoroutineDispatchers
+) {
+    suspend fun getTransactionsHistory(
         tokenPublicKey: String,
         signatures: List<String>
     ): List<HistoryTransaction> {
-        return withContext(Dispatchers.IO) {
-            getTransactionsFromCache(signatures)
-                .ifSizeNot(signatures.size) {
-                    fetchTransactionsFromRemote(signatures)
-                }
-                .let { transactions ->
-                    historyTransactionMapper.mapFromTransactionDetails(
-                        transactions = transactions,
-                        accountsInfo = getAccountsInfo(transactions),
-                        userPublicKey = tokenKeyProvider.publicKey,
-                        tokenPublicKey = tokenPublicKey
-                    )
-                }
-                .sortedByDescending { it.date.toInstant().toEpochMilli() }
+        return withContext(dispatchers.io) {
+            getTransactions(signatures, tokenPublicKey)
```

Comment:
COROUTINES EXPERTS, ATTENTION

now all the private methods are not `suspend` - they are working on IO thread that Repository declares here